### DISCUSSION TITLE
WithSql异常

### DISCUSSION THREAD

**Discussion by han1157** [dotnetcore/FreeSql#1267] [General]
var sql = "select * from haha where id=@id";
var rs = await _freeSql.Select<HahaModel>().WithSql(sql, new { id = "123456" }).ToListAsync();
难道只有我遇到这个问题，还是使用姿势不对
MySqlException: Parameter '@id' must be defined.

**Comment by han1157** [comment]
3.2.665版本

**Comment by 2881099** [comment]
你没有看 ado 文档。

如果用的 freesql.provider.mysql.dll：var sql = "select * from haha where id=?id";

如果用的 freesql.provider.mysqlconnector.dll：var sql = "select * from haha where id=@id";

因为两个引用的驱动不一样。

**Reply by han1157** [reply]
谢谢
